migrate/list-convert.go

```diff
@@ -0,0 +1,296 @@
+				fmt.Println("FSUUID found already, we can't convert this drive unless is unique: ", NewDrive.Status.FSUUID)
+			} else {
+				fmt.Println("element not found, FSUUID is unique, proceed to be added/appended for convertion")
+				setOfUniqueFSUUIDs[NewDrive.Status.FSUUID] = struct{}{}
+				// Append the new drive to the list
+				newDrives = append(newDrives, NewDrive)
+			}
+		}
+	}
+
+	// Return the new list of drives
+	return newDrives, nil
+}
+
+func main() {
+
+	// Get the kubernetes configuration file
+	kubeConfig, err := getKubeConfig()
+	if err != nil {
+		fmt.Printf("%s: Could not connect to kubernetes. %s=%s\n", "Error", "KUBECONFIG", kubeConfig)
+		os.Exit(1)
+	}
+
+	// Get the latest interface available
+	driveInterface, err := directcsiclient.DirectCSIDriveInterfaceForConfig(kubeConfig)
+	if err != nil {
+		fmt.Printf("%s: could not initialize drive adapter client: err=%v\n", "Error", err)
+		os.Exit(1)
+	}
+
+	// Get volume interface
+	volumeInterface, err := directcsiclient.DirectCSIVolumeInterfaceForConfig(kubeConfig)
+	if err != nil {
+		fmt.Printf("%s: could not initialize drive adapter client: err=%v\n", "Error", err)
+		os.Exit(1)
+	}
+
+	// Then we list the drives
+	// This is getting all the info from all the drives
+	ctx, _ := context.WithCancel(context.Background())
+	driveList, err := driveInterface.List(ctx, metav1.ListOptions{})
+	if err != nil {
+		fmt.Println("error")
+		os.Exit(1)
+	}
+	volumeList, err := volumeInterface.List(ctx, metav1.ListOptions{})
+	if err != nil {
+		fmt.Println("error")
+		os.Exit(1)
+	}
+
+	// Create volume list
+	filteredVolumes := []directcsi.DirectCSIVolume{}
+	for _, volume := range volumeList.Items {
+		// We currently don't filter volumes
+		filteredVolumes = append(filteredVolumes, volume)
+	}
+
+	// To filter the drives by status and version:
+	filteredDrives := []directcsi.DirectCSIDrive{}
+	for _, drive := range driveList.Items {
+		isDriveReadyOrInUse := false
+		isDrivev1beta5Version := false
+		isFilesystemUUIDPopulated := false
+		isPathInDevNotation := false
+		isDriveFinalizerCorrect := false
+		// Only for Ready and InUse status are valid here for this list of migration:
+		if drive.Status.DriveStatus == directcsi.DriveStatusReady || drive.Status.DriveStatus == directcsi.DriveStatusInUse {
+			isDriveReadyOrInUse = true
+		}
+		// Only for version v1beta5 is applicable the migration
+		groupAndVersion := strings.Split(drive.TypeMeta.APIVersion, `/`)
+		gvk, _ := directcsiclient.GetGroupKindVersions(
+			groupAndVersion[0],
+			drive.TypeMeta.Kind,
+			groupAndVersion[1],
+			directv1beta4.Version,
+			directv1beta3.Version,
+			directv1beta2.Version,
+			directv1beta1.Version,
+			directv1alpha1.Version,
+		)
+		if gvk.Group == "direct.csi.min.io" && gvk.Version == "v1beta5" {
+			isDrivev1beta5Version = true
+		} else {
+			isDrivev1beta5Version = false
+			fmt.Println("This drive's version cannot be migrated: ", gvk.Group, gvk.Version)
```

Comment:
we can support migration of objects >= v1beta3 (v2.0.9) and error out for versions < v1beta3